### PR TITLE
Added Calendar e2e tests and EqualDay specification tests

### PR DESCRIPTION
**Pull request by rdworak37**

### CODE REVIEW

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+				.nextSpecialOffer(OffsetDateTime.of(2019, 6, 5, 13, 19, 0, 0, ofHours(9)))
+				.build(em);
+		moeSzyslak = customer("Moe", "Szyslak")
+				.registrationDate(2015, 03, 02)
+				.birthDate(LocalDate.of(2002, 01, 21))
+				.lastOrderTime(LocalDateTime.of(2016, 8, 21, 5, 51,0))
+				.nextSpecialOffer(OffsetDateTime.of(2020, 6, 16, 16, 17, 0, 0, ofHours(9)))
+				.build(em);
+
+		joeQuimby = customer("Joe", "Quimby").golden().build(em);
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:68]: formatting should be consistent with the above customers

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+		assertThat(found).hasSize(2).containsOnly(homerSimpson, moeSzyslak);
+
+		EqualDay<Customer> secondDateOfNextSpecialOffer = new EqualDay<>(queryCtx, "dateOfNextSpecialOfferInstant", new String[] { "2019-06-05T09:34:13.000+04:00" }, defaultConverter);
+		found = customerRepo.findAll(secondDateOfNextSpecialOffer);
+
+		assertThat(found).hasSize(1).containsOnly(margeSimpson);
+	}
+
+	@Test
+	public void filterByInstantWithCustomDateFormat() {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:85]: filter -> filters

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+		assertThat(found).hasSize(2).containsOnly(homerSimpson, moeSzyslak);
+
+		EqualDay<Customer> secondDateOfNextSpecialOffer = new EqualDay<>(queryCtx, "dateOfNextSpecialOffer", new String[] { "2019-06-05T08:34:13.000+04:00" }, defaultConverter);
+		found = customerRepo.findAll(secondDateOfNextSpecialOffer);
+
+		assertThat(found).hasSize(1).containsOnly(margeSimpson);
+	}
+
+	@Test
+	public void filterByOffsetDateTimeWithCustomDateFormat() {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:129]: filter -> filters

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+		assertThat(found).hasSize(2).containsOnly(homerSimpson, margeSimpson);
+
+		EqualDay<Customer> secondBirthDate = new EqualDay<>(queryCtx, "birthDate", new String[] { "2002-01-21" }, defaultConverter);
+		found = customerRepo.findAll(secondBirthDate);
+
+		assertThat(found).hasSize(1).containsOnly(moeSzyslak);
+	}
+
+	@Test
+	public void filterByLocalDateWithCustomDateFormat() {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:151]: filter -> filters

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+		assertThat(found).hasSize(2).containsOnly(homerSimpson, moeSzyslak);
+
+		EqualDay<Customer> secondLastOrderTime = new EqualDay<>(queryCtx, "lastOrderTime", new String[] { "2012-08-11T02:34:19" }, defaultConverter);
+		found = customerRepo.findAll(secondLastOrderTime);
+
+		assertThat(found).hasSize(1).containsOnly(margeSimpson);
+	}
+
+	@Test
+	public void filterByLocalDateTimeWithCustomDateFormat() {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:173]: filter -> filters

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+		assertThat(found).hasSize(2).containsOnly(homerSimpson, margeSimpson);
+
+		EqualDay<Customer> secondRegistrationCalendar = new EqualDay<>(queryCtx, "registrationCalendar", new String[] { "2015-03-02" }, defaultConverter);
+		found = customerRepo.findAll(secondRegistrationCalendar);
+
+		assertThat(found).hasSize(1).containsOnly(moeSzyslak);
+	}
+
+	@Test
+	public void filterByCalendarWithCustomDateFormat() {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:195]: filter -> filters

File: src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+		assertThat(found).hasSize(2).containsOnly(homerSimpson, margeSimpson);
+
+		EqualDay<Customer> secondRegistrationDate = new EqualDay<>(queryCtx, "registrationDate", new String[] { "2015-03-02" }, defaultConverter);
+		found = customerRepo.findAll(secondRegistrationDate);
+
+		assertThat(found).hasSize(1).containsOnly(moeSzyslak);
+	}
+
+	@Test
+	public void filterByDateWithCustomDateFormat() {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/domain/EqualDayIntegrationTest.java:217]: filter -> filters

File: src/test/java/net/kaczmarzyk/e2e/converter/CalendarE2eTest.java [hunk 8]
```diff
@@ -46,16 +53,84 @@ public static class CalendarSpecController {
...
+        @ResponseBody
+        public Object findCustomersRegisteredGreaterOrEqual(
+                @Spec(path="registrationCalendar", params="registeredAfterOrEqualDate", spec=GreaterThanOrEqual.class) Specification<Customer> spec) {
+            return customerRepo.findAll(spec);
+        }
+
+        @RequestMapping(value = "/customers", params = "registeredAfterOrEqualDateCustomConfig")
+        @ResponseBody
+        public Object findCustomersRegisteredAfterOrEqualWithCustomConfigContainingTime(
+                @Spec(path="registrationCalendar", params="registeredAfterOrEqualDateCustomConfig", config="yyyy-MM-dd\'T\'HH:mm:ss", spec=GreaterThanOrEqual.class) Specification<Customer> spec) {
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/e2e/converter/CalendarE2eTest.java:133]: I suggest adding tests for `Equal` specification also

**Review by jradlica — APPROVED** [review]

File: src/test/java/net/kaczmarzyk/e2e/converter/CalendarE2eTest.java [hunk 9]
```diff
@@ -101,4 +191,221 @@ public void findsByEqualDayUsingCustomConfigIgnoringTime() throws Exception {
...
+    }
+
+    @Test
+    public void findsByEqualWithCustomConfigContainingTime() throws Exception {
+        customer("Adam", "Flayman")
+                .registrationDate(2014, 3, 15, 23, 59, 59, 999)
+                .build(em);
+        customer("Vanessa", "Bloom")
+                .registrationDate(2014, 3, 15, 12, 34, 19, 0)
+                .build(em);
```
Comment (gotfrydoh) [inline src/test/java/net/kaczmarzyk/e2e/converter/CalendarE2eTest.java:217]: I would extract saving test customers to `@Before` method so we are sure that during execution of all tests they are available. In current implementation not all tests use them (but they should). For example in `findsByBetween` you use default config for specifying date range - that's correct but in db we have only records with `registrationDate` with midnight time - we do not know if record with time part different than midnight will be returned. Maybe the part of tests name `CustomConfigContainingTime` was confusing but it refers only to format in which the date will be provided. Alternative way is to add customers with time part to other tests.

**Review by gotfrydoh — APPROVED** [review]

**Review by tkaczmarzyk — APPROVED** [review]

**Pull request merged**